src/PointwiseFunctions/GeneralRelativity/Python/Bindings.cpp

```diff
@@ -6,16 +6,189 @@
 #include <type_traits>
 
 #include "DataStructures/Tensor/TypeAliases.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Lapse.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ProjectionOperators.hpp"
 #include "PointwiseFunctions/GeneralRelativity/Psi4Real.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Ricci.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Shift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalOneForm.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalVector.hpp"
 
 namespace py = pybind11;
 
-namespace GeneralRelativity {
+namespace GeneralRelativity::py_bindings {
+
+namespace {
+template <size_t Dim, IndexType Index>
+void bind_christoffel_impl(py::module& m) {  // NOLINT
+  m.def("christoffel_first_kind",
+        static_cast<tnsr::abb<DataVector, Dim, Frame::Inertial, Index> (*)(
+            const tnsr::abb<DataVector, Dim, Frame::Inertial, Index>&)>(
+            &::gr::christoffel_first_kind),
+        py::arg("d_metric"));
+  m.def("christoffel_second_kind",
+        static_cast<tnsr::Abb<DataVector, Dim, Frame::Inertial, Index> (*)(
+            const tnsr::abb<DataVector, Dim, Frame::Inertial, Index>&,
+            const tnsr::AA<DataVector, Dim, Frame::Inertial, Index>&)>(
+            &::gr::christoffel_second_kind),
+        py::arg("d_metric"), py::arg("inverse_metric"));
+}
+
+void bind_christoffel(py::module& m) {  // NOLINT
+  bind_christoffel_impl<1, IndexType::Spatial>(m);
+  bind_christoffel_impl<2, IndexType::Spatial>(m);
+  bind_christoffel_impl<3, IndexType::Spatial>(m);
+  bind_christoffel_impl<1, IndexType::Spacetime>(m);
+  bind_christoffel_impl<2, IndexType::Spacetime>(m);
+  bind_christoffel_impl<3, IndexType::Spacetime>(m);
+}
+
+template <size_t Dim>
+void bind_lapse_impl(py::module& m) {  // NOLINT
+  m.def("lapse",
```

Comment:
Just put these all in one function like `bind_impl<Dim>`, then just call it with the three values for dim directly in `PYBIND11_MODULE` below. Also combine christoffel/ricci in one function and call directly in `PYBIND11_MODULE`. Reduces a lot of the boilerplate code in this file.